app/services/move_events/params_validator.rb

```diff
@@ -4,12 +4,22 @@ module MoveEvents
   class ParamsValidator
     include ActiveModel::Validations
 
-    attr_reader :timestamp, :type, :cancellation_reason, :rejection_reason
+    attr_reader :timestamp, :type, :date, :cancellation_reason, :rejection_reason
 
-    validates :type, presence: true, inclusion: { in: %w[cancel complete lockouts redirects reject] }
+    validates :type, presence: true, inclusion: { in: %w[cancel complete lockouts redirects approve reject] }
     validates :cancellation_reason, inclusion: { in: Move::CANCELLATION_REASONS }, if: -> { type == 'cancel' }
     validates :rejection_reason, inclusion: { in: Move::REJECTION_REASONS }, if: -> { type == 'reject' }
-    validates_each :timestamp, presence: true do |record, attr, value|
+
+    validates :date, presence: true, if: -> { type == 'approve' }
+    validates :timestamp, presence: true
+
+    validates_each :date, allow_nil: true do |record, attr, value|
+      Date.strptime(value, '%Y-%m-%d')
+    rescue ArgumentError
+      record.errors.add attr, 'is not a valid date.'
+    end
+
+    validates_each :timestamp, allow_nil: true do |record, attr, value|
```

Comment:
I tweaked this very slightly - the idea is that if we want to validate other `time` attributes they can be listed here, but we explicitly validate `presence` for each attribute depending on the event type. At the moment we don't have other `time` attributes but I still think it's a nice approach.